### PR TITLE
fix(server): fail closed on missing task harness ACL

### PR DESCRIPTION
**Pull request by chaliy**
### Motivation
- API-driven session-task executor calls must enforce the same harness→agent→session network ACL as in-turn execution and must not fail open when the harness layer is missing or unresolved.

### Description
- Resolve legacy `NULL` session `harness_id` via `org_init::base_harness_id` in `tool_context_for_ctx` so the normal base-harness fallback is applied before folding network ACL overlays.
- Change `load_harness_chain` to require a concrete `HarnessId` start and return an error when the leaf or a parent harness cannot be found, preventing an empty chain that would drop harness restrictions.
- Propagate the effective overlay ACL into the constructed `ToolContext` unchanged, so executors run under the derived ACL or the call fails closed.
- Add regression tests and test helpers that provision the built-in `base` harness and cover (1) NULL-harness sessions using the base harness and (2) sessions referencing a missing/stale harness which must error rather than produce an unrestricted `ToolContext`.

### Testing
- Ran `cargo fmt` / `cargo fmt --check` and applied formatting corrections; check passed after formatting.
- Ran focused tests `domains::session_tasks::commands::tests::tool_context_for_ctx` (three new/updated tests) with `cargo test -p everruns-server ... --all-features`, and all targeted tests passed (3 passed; 0 failed).
- Verified `git diff --check` and staged/committed the two modified files (`crates/server/src/domains/session_tasks/queries.rs`, `crates/server/src/domains/session_tasks/commands.rs`).

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a2c4f964ea4832b904fc1ac7d689389)

### CODE REVIEW

File: crates/server/src/domains/session_tasks/queries.rs [hunk 1]
```diff
@@ -161,17 +169,12 @@ pub async fn tool_context_for_ctx(
 /// Walk the harness parent chain (leaf → root), then reverse to root-to-leaf.
 ///
 /// Mirrors `DbHarnessStore::get_harness_chain` exactly, including cycle
-/// protection. Returns `Err` if a referenced parent is missing.
-/// Returns an empty vec if the session has no harness_id.
+/// protection. Returns `Err` if the leaf or a referenced parent is missing.
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/server/src/domains/session_tasks/queries.rs]: Fixed — reworded the docstring to call out the intentional divergence: it mirrors `DbHarnessStore::get_harness_chain`'s *cycle protection* but **fails closed**, returning `Err` for a missing leaf/parent where that helper yields an empty chain. (Rebased onto current `main` via cherry-pick, since the branch's old base diverged from the rewritten main.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/server/src/domains/session_tasks/queries.rs [hunk 2]
```diff
@@ -188,13 +191,12 @@ async fn load_harness_chain(
...
-            }
-            return Err(CommandError::internal(anyhow::anyhow!(
-                "Parent harness not found"
-            )));
+            let kind = if chain.is_empty() {
+                "Leaf harness"
+            } else {
+                "Parent harness"
+            };
+            return Err(CommandError::internal(anyhow::anyhow!("{kind} not found")));
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/server/src/domains/session_tasks/queries.rs]: Fixed — the error now includes the org and the harness that couldn't be loaded: `"{kind} not found (org_id={org_id}, harness_id={current_id})"`, so logs are actionable without changing control flow. (The `fails_closed` test still matches on `"Leaf harness not found"`, which remains a prefix.)

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/server/src/domains/session_tasks/commands.rs [hunk 3]
```diff
@@ -351,8 +351,43 @@ mod tests {
...
+        db: &Arc<StorageBackend>,
+        network_access: Option<NetworkAccessList>,
+    ) -> HarnessId {
+        if let Some(existing) = db
+            .get_harness_by_name(DEFAULT_ORG_ID, "base")
+            .await
+            .unwrap()
+        {
+            return existing.id;
+        }
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/server/src/domains/session_tasks/commands.rs:390]: Fixed — `ensure_base_harness` now asserts `existing.is_built_in` before returning the found row, so the helper truly guarantees the built-in base harness that `org_init::base_harness_id` resolves, and a stray user-created "base" harness fails the test loudly instead of silently diverging.

---
_Generated by [Claude Code](https://claude.ai/code)_
